core/field_checkbox.js

```diff
@@ -91,49 +94,98 @@ Blockly.FieldCheckbox.prototype.initView = function() {
       this.fieldGroup_);
   var textNode = document.createTextNode(Blockly.FieldCheckbox.CHECK_CHAR);
   this.checkElement_.appendChild(textNode);
-  this.checkElement_.style.display = this.state_ ? 'block' : 'none';
+  this.checkElement_.style.display = this.value_ ? 'block' : 'none';
+
+  if (this.borderRect_) {
+    this.borderRect_.setAttribute('width',
+        this.size_.width + Blockly.BlockSvg.SEP_SPACE_X);
+  }
 };
 
 /**
- * Return 'TRUE' if the checkbox is checked, 'FALSE' otherwise.
- * @return {string} Current state.
+ * Checkboxes have a constant width.
+ * @private
  */
-Blockly.FieldCheckbox.prototype.getValue = function() {
-  return String(this.state_).toUpperCase();
+Blockly.FieldCheckbox.prototype.render_ = function() {
+  this.size_.width = 5;
```

Comment:
Make a constant (Blockly.FieldCheckbox.WIDTH) to use here and in the constructor.

Reply:
Fixed